### PR TITLE
unit tests: do not fail when `/etc/machine-id` is empty

### PR DESCRIPTION
**Pull request by enr0n**
This is sort of a follow on to #25120, since some of these tests also fail when `/etc/machine-id` exists, but is empty. To simplify these checks, add a new test helper `machine_id_initialized()` which covers the cases when `/etc/machine-id` is empty, non-existent, or contains `"uninitialized"`.

For one test failure (`test-journal-interleaving`), it seemed the appropriate resolution was to actually patch the source. The rest of the changes are in unit tests that should be at least partially skipped when `/etc/machine-id` has not been initialized.

### CODE REVIEW

File: src/shared/tests.c [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #include "strv.h"
 #include "tests.h"
 #include "tmpfile-util.h"
+#include "id128-util.h"
```
Comment (bluca) [inline src/shared/tests.c]: sort alphabetically

File: src/libsystemd/sd-journal/journal-file.c [hunk 2]
```diff
@@ -502,11 +502,10 @@ static int journal_file_verify_header(JournalFile *f) {
                 int r;
 
                 r = sd_id128_get_machine(&machine_id);
-                if (r < 0)
-                        return r;
-
-                if (!sd_id128_equal(machine_id, f->header->machine_id))
+                if (r >= 0 && !sd_id128_equal(machine_id, f->header->machine_id))
                         return -EHOSTDOWN;
+                else if (r < 0 && !IN_SET(r, -ENOENT, -ENOMEDIUM, -ENOPKG))
```
Comment (yuwata) [inline src/libsystemd/sd-journal/journal-file.c]: Hm, IIUC the commit message, this is for testing. In real execution, I think the previous stricter check should be kept. How about to skip the test with a log message instead?

File: src/shared/tests.c [hunk 3]
```diff
@@ -360,3 +361,14 @@ const char *ci_environment(void) {
 
         return (ans = NULL);
 }
+
+int machine_id_initialized(void) {
+        sd_id128_t id;
+        int r;
+
+        r = id128_read("/etc/machine-id", ID128_FORMAT_PLAIN, &id);
```
Comment (yuwata) [inline src/shared/tests.c]: Why not simply use `sd_id128_get_machine()`?

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/shared/tests.c (hunk 3 above)
Comment (yuwata) [reply src/shared/tests.c]: See #25734.

File: src/libsystemd/sd-journal/journal-file.c (hunk 2 above)
Comment (enr0n) [reply src/libsystemd/sd-journal/journal-file.c]: Okay, that works for me.

File: src/shared/tests.c (hunk 1 above)
Comment (enr0n) [reply src/shared/tests.c]: Whoops, thanks!

File: src/shared/tests.c (hunk 3 above)
Comment (enr0n) [reply src/shared/tests.c]: Oh good point. I originally wrote this patch before your recent changes, and I wanted `ID128_PLAIN_OR_UNINIT`. I'll fix this.

Comment (yuwata) [reply src/shared/tests.c]: After #25734, you can simply write the check by `sd_id128_get_machine(NULL) >= 0`.

Comment (enr0n) [reply src/shared/tests.c]: Thanks, I rebased on your branch and will re-push once that is merged.

**Comment by enr0n** [comment]
I dropped the `machine_id_initialized()` helper in favor of using `sd_id128_get_machine(NULL)`, and I modified `test-journal-interleaving` rather than modifying the source. PTAL.

**Review by yuwata — APPROVED** [review]
